test_filenames_and_formats appears flaky

I'm trying to build dask 2.11.0 for Fedora. I did this, and then noticed I forgot to update some dependencies, so I ran it again. As there were 3 releases, that made for 6 builds. However, it failed twice in the first set, and twice in the second, so only 33% success. The failing builds were all about `test_filenames_and_formats`:
```
______ test_filenames_and_formats[mydask.pdf.svg-None-mydask.pdf.svg-SVG] ______
[gw27] linux -- Python 3.7.6 /usr/bin/python3
filename = 'mydask.pdf.svg', format = None, target = 'mydask.pdf.svg'
expected_result_type = <class 'IPython.core.display.SVG'>
    @pytest.mark.parametrize(
        "filename,format,target,expected_result_type",
        [
            pytest.param(
                "mydaskpdf", "svg", "mydaskpdf.svg", SVG, marks=ipython_not_installed_mark
            ),
            ("mydask.pdf", None, "mydask.pdf", type(None)),
            pytest.param(
                "mydask.pdf", "svg", "mydask.pdf.svg", SVG, marks=ipython_not_installed_mark
            ),
            pytest.param(
                "mydaskpdf", None, "mydaskpdf.png", Image, marks=ipython_not_installed_mark
            ),
            pytest.param(
                "mydask.pdf.svg",
                None,
                "mydask.pdf.svg",
                SVG,
                marks=ipython_not_installed_mark,
            ),
        ],
    )
    def test_filenames_and_formats(filename, format, target, expected_result_type):
        result = dot_graph(dsk, filename=filename, format=format)
>       assert os.path.isfile(target)
E       AssertionError: assert False
E        +  where False = <function isfile at 0xf785541c>('mydask.pdf.svg')
E        +    where <function isfile at 0xf785541c> = <module 'posixpath' from '/usr/lib/python3.7/posixpath.py'>.isfile
E        +      where <module 'posixpath' from '/usr/lib/python3.7/posixpath.py'> = os.path
dask/tests/test_dot.py:222: AssertionError
```
First go around:
* [Fedora Rawhide](https://koji.fedoraproject.org/koji/taskinfo?taskID=41672953)
* [Fedora 32](https://koji.fedoraproject.org/koji/buildinfo?buildID=1466675) works
* [Fedora 31](https://koji.fedoraproject.org/koji/taskinfo?taskID=41672950)
Second go around:
* [Fedora Rawhide](https://koji.fedoraproject.org/koji/taskinfo?taskID=41674670)
* [Fedora 32](https://koji.fedoraproject.org/koji/buildinfo?buildID=1466707) works
* [Fedora 31](https://koji.fedoraproject.org/koji/taskinfo?taskID=41674663)

Looking at the HW info, it doesn't appear to be arch-specific.

--- a/dask/tests/test_dot.py
+++ b/dask/tests/test_dot.py
@@ -217,11 +217,10 @@ def test_dot_graph_defaults():
         ),
     ],
 )
-def test_filenames_and_formats(filename, format, target, expected_result_type):
-    result = dot_graph(dsk, filename=filename, format=format)
-    assert os.path.isfile(target)
+def test_filenames_and_formats(tmpdir, filename, format, target, expected_result_type):
+    result = dot_graph(dsk, filename=str(tmpdir.join(filename)), format=format)
+    assert tmpdir.join(target).exists()
     assert isinstance(result, expected_result_type)
-    ensure_not_exists(target)
 
 
 def test_delayed_kwargs_apply():

